server/src/handlers.py

```diff
@@ -1,39 +1,104 @@
+    return web.json_response({'message': 'pong'})
 
 
-async def create_room(request: web.Request) -> web.Response:
+async def registration(request: web.Request) -> web.Response:
     data = await request.json()
-    some_room = room.Room(data)
+    print(type(data))
+    print(data)
+    user = User(data)
     try:
-        some_room.validate()
+        user.validate()
+        await db.create_user(request.app['pool'], user)
     except DataError as e:
         return web.json_response(e.to_primitive(), status=400)
-    if await db.room_is_occupied(data):
-        return web.json_response({'msg': 'room with that name already exists'}, status=409)
-    await db.create_room(data)
-    return web.json_response({'msg': 'OK'}, status=201)
-
-
-async def get_list_rooms(request: web.Request) -> web.Response:
-    page = int(request.query.get('page', '1'))
-    limit = int(request.query.get('limit', '10'))
-    rooms, total_count = await db.get_rooms(page, limit)
-    return web.json_response({
-        "rooms": rooms,
-        "paginator": {
-            "page": page,
-            "limit": limit,
-            "total_pages": math.ceil(total_count / limit),
-        }
-    }, status=200, dumps=parse_json)
+    except UniqueViolationError:
+        return web.json_response({'message': 'user with that name already exists'}, status=409)
+    return web.json_response({'message': 'OK'}, status=201)
+
+
+async def login(request: web.Request) -> web.Response:
+    data = await request.json()
+    remember = data.pop('remember', False)
+    user = User(data)
+    try:
+        user.validate()
+    except DataError as e:
+        return web.json_response(e.to_primitive(), status=400)
+    password = user.password
+    user = await db.get_user(request.app['pool'], user)
+    if user is None:
+        return web.json_response({'message': 'user don\'t exists'}, status=404)
+    if not is_same_hash(user.password, password):
+        return web.json_response({'message': 'wrong password'}, status=400)
+    response = web.json_response({'message': 'OK'}, status=200)
+    response.set_cookie(
+        name='token',
+        value=encrypt_jwt(id=user.id, username=user.username),
+        httponly=True,
+        max_age=None if remember else 3600 * 24 * 7
+    )
+    return response
+
+
+async def logout(request: web.Request) -> web.Response:
+    response = web.json_response({'message': 'OK'})
+    response.del_cookie(name='token')
+    return response
+
+
+@auth_required
+async def create_game(request: web.Request) -> web.Response:
+    game = await db.create_game(request.app['pool'], request.user)
+    return web.json_response(asdict(game), status=201)
+
+
+@auth_required
+async def login_game(request: web.Request) -> web.Response:
+    data = await request.json()
+    if data['current_state'] != State.PENDING.value:
+        return web.json_response({'message': 'invalid state'}, status=400)
+    game = await db.get_game(request.app['pool'], data['id'])
```

Comment:
`id` is must be taken from path https://github.com/qerdcv/ttt-online/pull/17/files#diff-19385f6e9c431027c8b759b6ee5dc055611711ca06dd4b75fd65a6b899470e20R13